src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -979,11 +1040,7 @@ private bool ScanNumericLiteral(ref TokenInfo info)
             }
             else
             {
-                while ((ch = TextWindow.PeekChar()) >= '0' && ch <= '9')
-                {
-                    _builder.Append(ch);
-                    TextWindow.AdvanceChar();
-                }
+                ScanNumericLiteralSingleInteger(_builder, ref underscoreInWrongPlace, isHex: false, isBinary: false);
```

Comment:
Why `isBinary: false`? It looks like it should be possible to have binary here.

In fact, why isn't this line
```cs
                ScanNumericLiteralSingleInteger(_builder, ref underscoreInWrongPlace, isHex: isHex, isBinary: isBinary);
```

and hoisted out of the if-then-else?